test/wdio/css-variables/cmp.test.tsx

```diff
@@ -0,0 +1,156 @@
+import './variables.css';
+
+import { Fragment, h } from '@stencil/core';
+import { render } from '@wdio/browser-runner/stencil';
+
+const css = `:root {
+  --global-background: black;
+  --global-color: white;
+  --body-background: grey;
+}
+body {
+  background: var(--body-background);
+  color: var(--global-color);
+}`;
+
+describe('css-variables', () => {
+  beforeEach(() => {
+    render({
+      template: () => (
+        <>
+          <style>{css}</style>
+
+          <css-variables-no-encapsulation></css-variables-no-encapsulation>
+          <css-variables-shadow-dom></css-variables-shadow-dom>
+          {/* This second instance will be used to ensure encapsulation between it and the first instance */}
+          <css-variables-shadow-dom></css-variables-shadow-dom>
+        </>
+      ),
+    });
+  });
+
+  describe('css-variables-no-encapsulation', () => {
+    it('uses class-local css variables to set the background and color', async () => {
+      const blackLocalElm = await $('.black-local');
+
+      await expect(blackLocalElm).toHaveText('No encapsulation: Black background');
+      await expect(blackLocalElm).toHaveStyle({
+        background: 'none 0% 0% auto repeat padding-box border-box scroll rgba(0, 0, 0, 1)',
+        color: 'rgba(255,255,255,1)',
+      });
+    });
+
+    it('uses global css variables to set the background, color, and font weight', async () => {
+      const blackGlobalElm = await $('.black-global');
+
+      await expect(blackGlobalElm).toHaveText('No encapsulation: Black background (global style)');
+      await expect(blackGlobalElm).toHaveStyle({
+        background: 'none 0% 0% auto repeat padding-box border-box scroll rgba(0, 0, 0, 1)',
+        color: 'rgba(255,255,255,1)',
+      });
+
+      const fontWeight = await blackGlobalElm.getCSSProperty('font-weight');
```

Comment:
AFAICT, this property isn't exposed by `toHaveStyle`, hence the separate assertion